app/models/box.rb

```diff
@@ -108,4 +108,12 @@ def unblind!
   def blind!
     update_columns(blinded: true)
   end
+
+  def original_batches
+    Batch.where(id: samples.map(&:original_batch_id).uniq)
+  end
+
+  def batches
+    batches + original_batches
+  end
```

Comment:
**question**: isn't this calling the `batches` method recursively (instead of calling the association)?

You might want to name it `all_batches` or maybe this isn't needed anymore?

Reply:
This is dead code, we're using now only `Sample#original_batch` and `SamplesReport#original_batches` after applying the previous comment :)